package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -69,249 +70,52 @@ not reflect in the results.
-
-        cdef ns_int i, j
-
-        for i in range(DIM):
-            dx[i] = other[i] - ref[i]
+from libc cimport math
 
-        if self.periodic:
-            for i in range(DIM-1, -1, -1):
-                while dx[i] > self.c_pbcbox.hbox_diag[i]:
-                    for j in range(i, -1, -1):
-                        dx[j] -= self.c_pbcbox.box[i][j]
+DEF END = -1
 
-                while dx[i] <= self.c_pbcbox.mhbox_diag[i]:
-                    for j in range(i, -1, -1):
-                        dx[j] += self.c_pbcbox.box[i][j]
-
-    cdef dreal fast_distance2(self, rvec a, rvec b) nogil:
-        """Distance calculation between two points
-        for both PBC and non-PBC aware calculations
+DEF XX = 0
+DEF XY = 3
+DEF YY = 4
+DEF XZ = 6
+DEF YZ = 7
+DEF ZZ = 8
 
-        Returns the distance obeying minimum
-        image convention if periodic is set to ``True`` while
-        instantiating the :class:`_PBCBox` object.
-        """
 
-        cdef drvec dx
-        self.fast_pbc_dx(a, b, dx)
-        return drvec_norm2(dx)
+cdef extern from "calc_distances.h" nogil:
+    void minimum_image(double* x, float* box, float* inverse_box)
+    void minimum_image_triclinic(double* dx, float* box)
 
-    cdef real[:, ::1] fast_put_atoms_in_bbox(self, real[:, ::1] coords) nogil:
-        """Shifts all ``coords`` to an orthogonal brick shaped box
 
-        All the coordinates are brought into an orthogonal
-        box. The box vectors for the brick-shaped box
-        are defined in ``fast_update`` method.
+cdef inline float fmax(float a, float b):
+    if a > b:
+        return a
+    else:
+        return b
 
-        """
 
-        cdef ns_int i, m, d, natoms
-        cdef real[:, ::1] bbox_coords
-
-        natoms = coords.shape[0]
-        with gil:
-            bbox_coords = coords.copy()
-
-        if self.periodic:
-            if self.is_triclinic:
-                for i in range(natoms):
-                    for m in range(DIM - 1, -1, -1):
-                        while bbox_coords[i, m] < 0:
-                            for d in range(m+1):
-                                bbox_coords[i, d] += self.c_pbcbox.box[m][d]
-                        while bbox_coords[i, m] >= self.c_pbcbox.box[m][m]:
-                            for d in range(m+1):
-                                bbox_coords[i, d] -= self.c_pbcbox.box[m][d]
-            else:
-                for i in range(natoms):
-                    for m in range(DIM):
-                        while bbox_coords[i, m] < 0:
-                            bbox_coords[i, m] += self.c_pbcbox.box[m][m]
-                        while bbox_coords[i, m] >= self.c_pbcbox.box[m][m]:
-                            bbox_coords[i, m] -= self.c_pbcbox.box[m][m]
+cdef inline float fmin(float a, float b):
+    if a < b:
+        return a
+    else:
+        return b
 
-        return bbox_coords
+cdef inline float degsin(float deg):
+    # sin in degrees
+    if (deg > 90):
+        deg = 180 - deg
```

Comment:
Just out of curiosity: why do you need to transform the angle? In general sin(π -x) = sin(x). I assume 0 ≤ x < π (*is this true?*) so you'd always have sin(x) ≥ 0. This only makes a difference if you can have x outside the range but then I don't quite understand what the ultimate goal of this function is.

Reply:
Just me being bad at trig :)